x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/step_about_rule_details/index.test.tsx

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import { EuiProgress, EuiButtonGroup } from '@elastic/eui';
+import { ThemeProvider } from 'styled-components';
+import euiDarkVars from '@elastic/eui/dist/eui_theme_light.json';
+
+import { StepAboutRuleToggleDetails } from './';
+import { mockAboutStepRule } from '../../all/__mocks__/mock';
+import { HeaderSection } from '../../../../../components/header_section';
+import { StepAboutRule } from '../step_about_rule/';
+import { AboutStepRule } from '../../types';
+
+const theme = () => ({ eui: euiDarkVars, darkMode: true });
+
+describe('StepAboutRuleToggleDetails', () => {
+  let mockRule: AboutStepRule;
+
+  beforeEach(() => {
+    // jest carries state between mocked implementations when using
+    // spyOn. So now we're doing all three of these.
+    // https://github.com/facebook/jest/issues/7136#issuecomment-565976599
+    jest.resetAllMocks();
+    jest.restoreAllMocks();
+    jest.clearAllMocks();
+
+    mockRule = mockAboutStepRule();
+  });
+
+  test('it renders loading component when `loading` is true', () => {
+    const wrapper = shallow(
+      <StepAboutRuleToggleDetails
+        loading={true}
+        stepDataDetails={{
+          note: mockRule.note,
+          description: mockRule.description,
+        }}
+        stepData={mockRule}
+      />
+    );
+
+    expect(wrapper.find(EuiProgress).exists()).toBeTruthy();
+    expect(wrapper.find(HeaderSection).exists()).toBeTruthy();
+  });
+
+  test('it does not render details if stepDataDetails is null', () => {
+    const wrapper = shallow(
+      <StepAboutRuleToggleDetails loading={true} stepDataDetails={null} stepData={mockRule} />
+    );
+
+    expect(wrapper.find(StepAboutRule).exists()).toBeFalsy();
+  });
+
+  test('it does not render details if stepData is null', () => {
+    const wrapper = shallow(
+      <StepAboutRuleToggleDetails
+        loading={true}
+        stepDataDetails={{
+          note: '',
+          description: '',
+        }}
+        stepData={null}
+      />
+    );
+
+    expect(wrapper.find(StepAboutRule).exists()).toBeFalsy();
+  });
+
+  describe('note value does NOT exist', () => {
+    test('it does not render toggle buttons', () => {
+      const mockAboutStepWithoutNote = {
+        ...mockRule,
+        note: '',
+      };
```

Comment:
This description and variable names are slightly misleading. You are saying you are testing `note value does NOT exist`

However, you look to be testing a note that is an empty string.

You should do this if you want to omit the note:

```ts
const { note, ...mockAboutStepWithoutNote } = { ...mockRule };
```

That will omit the note rather than just setting it to an empty note. I would change the description or the implementation to be accurate that you are testing a note value which is an empty string.

Reply:
You're right, that's confusing there. I updated the language to reflect that I wanted to test when note is an empty string.